modules/carto/src/layers/schema/carto-raster-tile-loader.ts

```diff
@@ -50,13 +50,21 @@ function parseCartoRasterTile(
 ): Raster | null {
   if (!arrayBuffer) return null;
   const tile = parsePbf(arrayBuffer, TileReader);
-  const {bands, blockHeight, blockWidth} = tile;
+  const {bands, blockSize, blockWidth, blockHeight} = tile.bands;
+
+  if (!blockSize) {
+    assert(
+      blockWidth === blockHeight,
+      `blockWidth (${blockWidth}) must equal blockHeight (${blockHeight})`
+    );
+  }
+
   const numericProps = {};
   for (let i = 0; i < bands.length; i++) {
     const {name, data} = bands[i];
     numericProps[name] = data;
   }
-  return {blockWidth, blockHeight, cells: {numericProps, properties: []}};
+  return {blockSize: blockSize || blockWidth, cells: {numericProps, properties: []}};
```

Comment:
Just to be safe? Don't know why it would ever be 0, but maybe for an empty tile?
```suggestion
  return {blockSize: blockSize ?? blockWidth, cells: {numericProps, properties: []}};
```